src/types/result.ts

```diff
@@ -2,10 +2,7 @@ export type resultOverrides = {
   [flagId: string]: { heading?: string; description?: string };
 };
 
-export type ResultData = (
-  flagSet?: string,
-  overrides?: resultOverrides
-) => {
+export type ResultData = {
```

Comment:
This type is less complex to work with if it references the output directly rather than having to reference the more indirect `ReturnType<ResultData>`

Reply:
I agree with the simplification here - having a complex type and then referencing both it's `ReturnType<>` and `Parameters<>` is clever, but not necessarily that easy to follow.